metadata-db-app/src/migrations/058_add_provider_id_to_variables_table.clj

```diff
@@ -0,0 +1,26 @@
+(ns migrations.058-add-provider-id-to-variables-table
+  "Adds `provider_id` column to variables table."
+  (:require
+   [clojure.java.jdbc :as j]
+   [cmr.metadata-db.services.provider-validation :as provider-validation]
+   [config.mdb-migrate-helper :as h]))
+
+(defn up
+  "Migrates the database up from version 57 to 58."
+  []
+  (println "migrations.058-add-provider-id-to-variables-table up...")
+  ;; At this point in time, UMM-Vars support is still being developed, so we
+  ;; don't care about what's in the database right now, and the new field
+  ;; can't be null, so:
+  (h/sql "TRUNCATE TABLE METADATA_DB.cmr_variables")
+  (h/sql "TRUNCATE TABLE METADATA_DB.cmr_variable_associations")
+  (h/sql (format (str "ALTER TABLE METADATA_DB.cmr_variables "
+                      "ADD provider_id VARCHAR(%s) NOT NULL")
+                 provider-validation/PROVIDER_ID_MAX_LENGTH))
+  (h/sql "CREATE INDEX variables_p_vn ON METADATA_DB.cmr_variables (provider_id, variable_name)"))
```

Comment:
It looks like my comment about our indexes never made it through. I just want to double-check what query this index is supporting. When we search by variable name are we going to also pass the provider-id?

I think this existing index should be dropped and we should add provider-id to it (I'm assuming searching for variables in metadata-db will include the provider-id):
`(h/sql "CREATE INDEX variables_crdi ON METADATA_DB.cmr_variables (concept_id, revision_id, deleted)")`

Having said that it probably makes sense to add a database migration changing the indexes in the same PR as the updates to metadata-db searching for variables now that we include provider-id. Are we planning to do that as part of CMR-4239 or CMR-4240?